src/System.Runtime/tests/System/GC.cs

```diff
@@ -226,4 +226,84 @@ private class TestObject
             }
         }
     }
-}
\ No newline at end of file
+
+    [Fact]
+    public static void ReRegisterForFinalize()
+    {
+        ReRegisterForFinalizeTest.Run();
+    }
+
+    private class ReRegisterForFinalizeTest
+    {
+        public static void Run()
+        {
+            TestObject.Finalized = false;
+            CreateObject();
+
+            GC.Collect();
+            GC.WaitForPendingFinalizers();
+
+            Assert.True(TestObject.Finalized);
+        }
+
+        private static void CreateObject()
+        {
+            using (var obj = new TestObject())
+            {
+                GC.SuppressFinalize(obj);
+            }
+        }
+
+        private class TestObject : IDisposable
+        {
+            public static bool Finalized { get; set; }
+
+            ~TestObject()
+            {
+                Finalized = true;
+            }
+
+            public void Dispose()
+            {
+                GC.ReRegisterForFinalize(this);
+            }
+        }
+    }
+
+    [Fact]
+    public static void GetTotalMemoryTest_NoForceCollection()
+    {
+        GC.Collect();
+
+        byte[] bytes = new byte[50000];
+        int genBefore = GC.GetGeneration(bytes);
+
+        Assert.True(GC.GetTotalMemory(false) >= bytes.Length);
+        Assert.True(GC.GetGeneration(bytes) == genBefore);
```

Comment:
I would change the last line to

`Assert.True(GC.GetGeneration(bytes) >= genBefore);`

If you run this test in a simple way, eg, you run it in a standalone exe with all the other simple tests and in a sequential fashion (in other words, if you were to do a GC.Collect/WaitForPending/Colllect sequence, all the memory from the last test would be gone). Then this test is ok...However, as soon as these conditions are not true (eg, if you run this test with a bunch of other tests at the same time, say if we had an infrastructure that just picked random tests and called their Run method on multiple threads in a process) then this test would be meaningless 'cause you can never predict the generation. It could be the same, larger or smaller than before (there could be a GC that happens between new and the 1st GetGeneration, so genBefore could be >0).